web/base_viewer.js

```diff
@@ -735,6 +735,11 @@ class BaseViewer {
       if (!currentPage) {
         return;
       }
+      const factor =
+        this.spreadMode === SpreadMode.NONE || this._isScrollModeHorizontal
+          ? 1
+          : 2;
```

Comment:
I don't believe that the above condition is correct in *all* situations, what about e.g. PresentationMode and/or the `PDFSinglePageViewer`?

To address all of this, let's add a new (private) getter above, and use that instead, e.g. something like:
```javascript
  /**
   * @private
   */
  get _pageWidthScaleFactor() {
    if (
      this.spreadMode !== SpreadMode.NONE &&
      this.scrollMode !== ScrollMode.HORIZONTAL &&
      !this.isInPresentationMode()
    ) {
      return 2;
    }
    return 1;
  }
```
and also the following dummy in `web/pdf_single_page_viewer.js`:
```javascript
  get _pageWidthScaleFactor() {
    return 1;
  }
```